### PR TITLE
Change default work remaining to 0.0 instead of 1.0

### PR DESCRIPTION
**Pull request by johnjcasey**
This is to enable downscaling, and a default work remaining of 0 instead of 1 will not significantly impact upscaling

### CODE REVIEW

**Comment by johnjcasey** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by scwhittle** [comment]
IIUC this is just to ensure that we don't report a backlog size for the SDF which happens if hasProgress isn't implemented?
If so, could we change the behavior of without hasProgress from returning 1.0 to 0 in  invokeGetSize here
sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvoker.java ?

**Comment by johnjcasey** [comment]
Run Java PreCommit

**Comment by johnjcasey** [comment]
Run Java PreCommit

**Comment by johnjcasey** [comment]
Run Java PreCommit

**Comment by johnjcasey** [comment]
Both failing tests appear to be unrelated flakes

**Comment by bvolpato** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by bvolpato** [comment]
Run Java_Pulsar_IO_Direct PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java [hunk 1]
```diff
@@ -432,7 +432,7 @@ public static <InputT, OutputT> double invokeGetSize(
             .getProgress()
             .getWorkRemaining();
       } else {
-        return 1.0;
+        return 0.0;
```
Comment (bvolpato) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java:435]: Maybe adding a comment with the rationale for returning 0 will be useful for the next person

**Review by bvolpato — APPROVED** [review]

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java (hunk 1 above)
Comment (scwhittle) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java:435]: Please update the comment where the interface is defined, as it explicitly states 1.0.  The comment requested above would be better there too.

Comment (scwhittle) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java:435]: I'm referring to the DoFnInvoker.invokeGetSize comment

**Review by bvolpato — APPROVED** [review]

**Pull request merged**